packages/caip/src/utils.ts

```diff
@@ -14,6 +16,22 @@ export const isValidChainPartsPair = (
   chainReference: ChainReference,
 ) => constants.VALID_CHAIN_IDS[chainNamespace]?.includes(chainReference) || false
 
+export const generateAssetIdFromCosmosDenom = (denom: string): AssetId => {
```

Comment:
could make this more generic and pass in the native `assetId` to derive ibc or whatever else on other cosmossdk chains as well

Reply:
92cad99ad83d4c3c61ecdb14df87868b00f003d4